### PR TITLE
Consistently export error types thrown by msal-browser

### LINKED ISSUE #4115: MSAL inconsistently exports error types
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.17.0

### Wrapper Library

Not Applicable

### Wrapper Library Version

None

### Description

I'm writing error handling code around MSAL error constructs but am finding that the error types MSAL can throw outside of its boundaries are inconsistently exported. I'd like to tightly couple to them wherever possible in order to make sure we are properly addressing, identifying and adapting to MSAL errors now and over time. This is easy with some error types like `BrowserAuthError` of which `msal-browser` exports both the class and error codes. However, with other error types sometimes only the class is exported (`InteractionRequiredAuthError`), and for some error types, nothing is exported at all (`ServerError` , `ClientAuthError`).
 
It seems like any error that can be thrown outside of MSAL's boundary should have supporting types. Right now we've having to loosely couple based on error codes and I know for certain we're missing and not properly identifying all possible error types. Thanks!

### Error Message

_No response_

### Msal Logs

_No response_

### MSAL Configuration

```javascript
n/a (library types export issue)
```

### Relevant Code Snippets

```javascript
n/a (library types export issue)
```

### Reproduction Steps

View exports from `msal-browser` and/or try to access error classes and types in consuming software

### Expected Behavior

Any error type that MSAL can throw outside of its boundaries should be exported by MSAL.

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

Other

### Regression

_No response_

### Source

Internal (Microsoft)

### PR DESCRIPTION
**Pull request by JasonGore**
Consistently exports error types thrown by `msal-browser` per #4115.

Resolves #4115

### CODE REVIEW

File: lib/msal-browser/src/index.ts [hunk 1]
```diff
@@ -48,8 +48,12 @@ export {
     AuthenticationResult,
     // Error
     InteractionRequiredAuthError,
+    InteractionRequiredAuthErrorMessage,
     AuthError,
     AuthErrorMessage,
+    ClientAuthError,
+    ClientAuthErrorMessage,
+    ServerError,
```
Comment (JasonGore) [inline lib/msal-browser/src/index.ts:56]: I am not sure if [the `ClientConfigurationError` error class](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/src/error/ClientConfigurationError.ts) should also be exported. For now I'm assuming it's an internal error and not exporting it.

Comment (tnorling) [reply lib/msal-browser/src/index.ts:56]: Yes it should also be exported. Can we also have these exported from `msal-node`, if not already?

Comment (JasonGore) [reply lib/msal-browser/src/index.ts:58]: Updated.

File: change/@azure-msal-browser-5d9d1587-039a-4462-8dd3-b48d09664add.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (tnorling) [inline change/@azure-msal-browser-5d9d1587-039a-4462-8dd3-b48d09664add.json:2]:
```suggestion
  "type": "patch",
```

File: change/@azure-msal-common-7aa16f1c-8c7e-45db-b518-f5dbb0c48ea3.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (tnorling) [inline change/@azure-msal-common-7aa16f1c-8c7e-45db-b518-f5dbb0c48ea3.json:2]:
```suggestion
  "type": "patch",
```

File: change/@azure-msal-node-c27ce924-c732-41ee-adff-ed3093aafbed.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (tnorling) [inline change/@azure-msal-node-c27ce924-c732-41ee-adff-ed3093aafbed.json:2]:
```suggestion
  "type": "patch",
```

File: change/@azure-msal-browser-5d9d1587-039a-4462-8dd3-b48d09664add.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+  "comment": "Consistently export error types and messages for errors thrown by MSAL.",
```
Comment (tnorling) [inline change/@azure-msal-browser-5d9d1587-039a-4462-8dd3-b48d09664add.json:3]:
```suggestion
  "comment": "Consistently export error types and messages for errors thrown by MSAL #4117",
```

File: change/@azure-msal-common-7aa16f1c-8c7e-45db-b518-f5dbb0c48ea3.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+  "comment": "Consistently export error types and messages for errors thrown by MSAL.",
```
Comment (tnorling) [inline change/@azure-msal-common-7aa16f1c-8c7e-45db-b518-f5dbb0c48ea3.json:3]:
```suggestion
  "comment": "Consistently export error types and messages for errors thrown by MSAL #4117",
```

File: change/@azure-msal-node-c27ce924-c732-41ee-adff-ed3093aafbed.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+  "comment": "Consistently export error types and messages for errors thrown by MSAL.",
```
Comment (tnorling) [inline change/@azure-msal-node-c27ce924-c732-41ee-adff-ed3093aafbed.json:3]:
```suggestion
  "comment": "Consistently export error types and messages for errors thrown by MSAL #4117",
```

**Review by tnorling — APPROVED** [review]
Some changelog nits but otherwise looks good. Thanks for opening a PR!

**Comment by JasonGore** [comment]
@tnorling  My rationale for doing `minor` was the fact that the public API has changed, but is backwards-compatible. Just wanted to throw that out there before I accept these changes

**Review by peterzenz — APPROVED** [review]

**Review by jasonnutter — APPROVED** [review]

**Comment by jasonnutter** [comment]
Looks great, thanks for the PR!

**Pull request merged**